src/routes/_surfaces/workspace/explore/time-range-constructor/timeRangeUtils.ts

```diff
@@ -0,0 +1,589 @@
+    dateFormatOptions
+  )} - ${end.toLocaleDateString(undefined, dateFormatOptions)}`;
+};
+
+export const formatDateByInterval = (
+  interval: string, // DuckDB interval
+  date: string
+): string => {
+  switch (interval) {
+    case "5 minute":
+      return new Date(date).toLocaleDateString(undefined, {
+        year: "numeric",
+        month: "short",
+        day: "numeric",
+        hour: "numeric",
+        minute: "numeric",
+      });
+    case "15 minute":
+      return new Date(date).toLocaleDateString(undefined, {
+        year: "numeric",
+        month: "short",
+        day: "numeric",
+        hour: "numeric",
+        minute: "numeric",
+      });
+    case "1 hour":
+      return new Date(date).toLocaleDateString(undefined, {
+        year: "numeric",
+        month: "short",
+        day: "numeric",
+        hour: "numeric",
+      });
+    case "1 day":
+      return new Date(date).toLocaleDateString(undefined, {
+        year: "numeric",
+        month: "short",
+        day: "numeric",
+      });
+    case "7 day":
+      return new Date(date).toLocaleDateString(undefined, {
+        year: "numeric",
+        month: "short",
+        day: "numeric",
+      });
+    case "1 month":
+      return new Date(date).toLocaleDateString(undefined, {
+        year: "numeric",
+        month: "short",
+      });
+    case "1 year":
+      return new Date(date).toLocaleDateString(undefined, {
+        year: "numeric",
+      });
+    default:
+      throw new Error(`Unknown interval: ${interval}`);
+  }
+};
+
+export const prettyTimeGrain = (timeGrain: TimeGrain): string => {
+  if (!timeGrain) return "";
+  switch (timeGrain) {
+    // case TimeGrain.FiveMinutes:
+    //   return "5 minute";
+    // case TimeGrain.FifteenMinutes:
+    //   return "15 minute";
+    case TimeGrain.OneHour:
+      return "hourly";
+    case TimeGrain.OneDay:
+      return "daily";
+    // case TimeGrain.OneWeek:
+    //   return "weekly";
+    case TimeGrain.OneMonth:
+      return "monthly";
+    case TimeGrain.OneYear:
+      return "yearly";
+    default:
+      throw new Error(`Unknown time grain: ${timeGrain}`);
+  }
+};
+
+const getTimeRangeDuration = (
+  timeRangeName: TimeRangeName,
+  allTimeRange: TimeSeriesTimeRange
+): number => {
+  switch (timeRangeName) {
+    // case TimeRangeName.LastHour:
+    //   return 60 * 60 * 1000;
+    case TimeRangeName.Last6Hours:
+      return 6 * 60 * 60 * 1000;
+    case TimeRangeName.LastDay:
+      return 24 * 60 * 60 * 1000;
+    case TimeRangeName.Last2Days:
+      return 2 * 24 * 60 * 60 * 1000;
+    case TimeRangeName.Last5Days:
+      return 5 * 24 * 60 * 60 * 1000;
+    case TimeRangeName.LastWeek:
+      return 7 * 24 * 60 * 60 * 1000;
+    case TimeRangeName.Last2Weeks:
+      return 2 * 7 * 24 * 60 * 60 * 1000;
+    case TimeRangeName.Last30Days:
+      return 30 * 24 * 60 * 60 * 1000;
+    case TimeRangeName.Last60Days:
+      return 60 * 24 * 60 * 60 * 1000;
+    case TimeRangeName.AllTime:
```

Comment:
If we wanted to get custom time ranges -> time grain sets, it starts with casing `TimeRangeName.AllTime` to also have the logic `TimeRangeName.Custom`, right?

Seemsl ike we're not far off from custom time range / grain selection, but given the time this seems not possible for us to land in this PR

Reply:
Yeah, that's the start of the implementation. I think it's a bigger project though, so we should handle it in a different Issue and won't happen by EOD Tuesday regardless.